src/slider/interfaces.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { BaseComponentProps } from '../internal/base-component';
+import { FormFieldValidationControlProps } from '../internal/context/form-field-context';
+import { NonCancelableEventHandler } from '../internal/events';
+
+export interface SliderProps extends BaseComponentProps, FormFieldValidationControlProps {
+  /**
+   * Indicates the current value.
+   */
+  value?: number;
+
+  /**
+   * Indicates the minimum value.
+   */
+  min: number;
+
+  /**
+   * Indicates the maximum value.
+   */
+  max: number;
+
+  /**
+   * Formats the values. This will format both the labels and the tooltip.
+   */
+  valueFormatter?: (value: number) => string;
+
+  /**
+   * onChange handler.
+   */
+  onChange?: NonCancelableEventHandler<SliderProps.ChangeDetail>;
+
+  /**
+   * How big the step size is.
+   */
+  step?: number;
+
+  /**
+   * Hides the colored fill line, so only the handle is visible.
+   */
+  hideFillLine?: boolean;
+
+  /**
+   * Show the tick marks along the axis. Use with stepped sliders, except in extreme cases.
+   */
+  tickMarks?: boolean;
+
+  /**
+   * Whether or not the slider is disabled.
+   */
+  disabled?: boolean;
+
+  /**
+   * Labels shown between the minimum and maximum values.
+   */
+  referenceValues?: Array<number>;
```

Comment:
It is better to accept `ReadonlyArray<number>` to guarantee that we are not going to update it.

Do we expect the reference values array to be sorted? Can it include values outside of the min/max range?

Reply:
It doesn't have to be sorted, since the calculation just takes the value itself and places it in that column along the slider. If its outside the min / max, it just isn't shown.

Reply:
Added a test for that, by the way.